addons/website/static/src/interactions/header/base_header_special.js

```diff
@@ -0,0 +1,148 @@
+import { BaseHeader } from "@website/interactions/header/base_header";
+
+export class BaseHeaderSpecial extends BaseHeader {
+    dynamicContent = {
+        ...this.dynamicContent,
+        ".o_header_hide_on_scroll .dropdown-toggle": {
+            "t-on-show.bs.dropdown": this.onDropdownShow,
+        },
+        ".o_header_hide_on_scroll :not(.modal-content) > .o_searchbar_form": {
+            "t-on-input": this.onSearchbarInput,
+        },
+    }
+
+    setup() {
+        super.setup();
+        this.isAnimated = false;
+
+        this.position = 0;
+        this.checkpoint = 0;
+        this.scrollOffset = 200;
+        this.scrollingDownward = true;
+
+        this.searchbarEl = this.el.querySelector(".o_header_hide_on_scroll :not(.modal-content) > .o_searchbar_form");
```

Comment:
Since you store it here and use it in dynamicContent, you could also make it a `dynamicSelector`, so you don't have to repeat the selector

Reply:
also, it's a bit weird now that it's outside of the condition (`if (this.hiddenOnScrollEl)`) that its name is as generic as `this.searchbarEl` with such a specific selector.
(Maybe that's just me...)

Reply:
I don't think it's a big issue. It is the same searchbar in all template, it's just that we want to store it only when nested inside the `hide_on_scroll` element. 
But yes, using the dynamicSelector would be better / clearer.